block_function.rb

```diff
@@ -0,0 +1,18 @@
+def do_calc(a, b)
+  yield
+end
+
+a = 7
```

Comment:
declaring these here is actually making your program work, but not in the expected way